### PR TITLE
1. Added category field for vector-layers 2. Refactored the layers caching filename creation and extraction into a separate class.

### PR DESCRIPTION
**Pull request by BarAgam**
1. Added category field for vector-layers
2. Refactored the layers caching filename creation and extraction into a separate class.

### CODE REVIEW

File: app/src/test/java/com/teamagam/gimelgimel/app/integration/interactors/DisplayVectorLayerInteractorTest.java [hunk 1]
```diff
@@ -148,11 +148,16 @@ private Scheduler createTestScheduler() {
     }
 
     private VectorLayer createVectorLayer1() {
```
Comment (teamagam5) [inline app/src/test/java/com/teamagam/gimelgimel/app/integration/interactors/DisplayVectorLayerInteractorTest.java]: I liked that refactor, but now I find no good reason for these two methods to exist. I think both should be inlined.

File: data/src/main/java/com/teamagam/gimelgimel/data/layers/LayersLocalCacheData.java [hunk 2]
```diff
@@ -80,51 +76,19 @@ private URI downloadToCache(VectorLayer vectorLayer, URL url) {
...
-        nameElements.add(VERSION_POSITION, String.valueOf(vectorLayer.getVersion()));
-        return TextUtils.join(NAME_SEPARATOR, nameElements) + KML_EXTENSION;
-    }
-
     private List<VectorLayer> extractVectorLayersFromFiles(File[] vectorLayerFiles) {
         List<VectorLayer> vectorLayers = new ArrayList<>(vectorLayerFiles.length);
         for (File file : vectorLayerFiles) {
             try {
-                vectorLayers.add(extractVectorLayerFromFile(file));
+                vectorLayers.add(mLayerFilenameSerializer.fromFilename(file.getName()));
```
Comment (teamagam5) [inline data/src/main/java/com/teamagam/gimelgimel/data/layers/LayersLocalCacheData.java]: I think toVectorLayer(String filename) is better than fromFilename(String filename). The "from filename" fact is already encapsulated in the parameter...

Comment (teamagam5) [reply data/src/main/java/com/teamagam/gimelgimel/data/layers/LayersLocalCacheData.java]: I concluded this after considering the extraction of a variable out of "mLayerFilenameSerializer.fromFilename(file.getName())"

File: data/src/main/java/com/teamagam/gimelgimel/data/message/adapters/MessageDataMapper.java [hunk 3]
```diff
@@ -84,8 +84,13 @@ public MessageData transformToData(Message message) {
      * @param message Object to be transformed.
      * @return {@link com.teamagam.gimelgimel.domain.messages.entity.Message} if valid {@link MessageData} otherwise null.
      */
-    public Message transform(MessageData message) {
-        return new MessageFromDataTransformer().transformFromData(message);
+    public Message tryTransform(MessageData message) {
+        try {
+            return new MessageFromDataTransformer().transformFromData(message);
+        } catch (Exception ex) {
```
Comment (teamagam5) [inline data/src/main/java/com/teamagam/gimelgimel/data/message/adapters/MessageDataMapper.java:90]: ex -> e ?
Petty as usual

File: data/src/main/java/com/teamagam/gimelgimel/data/message/adapters/MessageDataMapper.java [hunk 4]
```diff
@@ -197,14 +202,15 @@ public void visit(MessageAlertData message) {
...
-                    convertSeverity(content.getSeverity()));
+                    convertSeverity(content.getSeverity()), convertCategory(content.getCategory()));
         }
 
         private VectorLayer.Severity convertSeverity(String severity) {
-            if (VectorLayer.Severity.REGULAR.name().equalsIgnoreCase(severity)) {
-                return VectorLayer.Severity.REGULAR;
-            }
-            return VectorLayer.Severity.IMPORTANT;
+            return VectorLayer.Severity.parseCaseInsensitive(severity);
```
Comment (teamagam5) [inline data/src/main/java/com/teamagam/gimelgimel/data/message/adapters/MessageDataMapper.java:209]: Cool!

File: domain/src/main/java/com/teamagam/gimelgimel/domain/layers/LoadAllCachedLayersInteractor.java [hunk 5]
```diff
@@ -36,6 +36,13 @@ public LoadAllCachedLayersInteractor(ThreadExecutor threadExecutor,
     private Observable<VectorLayer> createObservable() {
         return Observable.just(null)
                 .flatMapIterable(x -> mLayersLocalCache.getAllCachedLayers())
+                .map(this::recreateAsUnimportant)
                 .doOnNext(vl -> mProcessNewVectorLayerInteractorFactory.create(vl, null).execute());
     }
+
+    private VectorLayer recreateAsUnimportant(VectorLayer vectorLayer) {
```
Comment (teamagam5) [inline domain/src/main/java/com/teamagam/gimelgimel/domain/layers/LoadAllCachedLayersInteractor.java:43]: Maybe this should reside in VectorLayer as a static method. Unless you consider it a logic operation? It's like a dedicated builder

File: domain/src/main/java/com/teamagam/gimelgimel/domain/messages/SendMessageInteractor.java [hunk 6]
```diff
@@ -39,7 +39,6 @@ public SendMessageInteractor(
                         .map(x -> createMessage())
                         .doOnNext(m -> mMessageNotifications.sending())
                         .flatMap(mMessagesRepository::sendMessage)
-                        .doOnNext(mMessagesRepository::putMessage)
```
Comment (teamagam5) [inline domain/src/main/java/com/teamagam/gimelgimel/domain/messages/SendMessageInteractor.java:42]: WTH was this doing here?!

**Review by teamagam5 — APPROVED** [review]

File: app/src/test/java/com/teamagam/gimelgimel/app/integration/interactors/DisplayVectorLayerInteractorTest.java (hunk 1 above)
Comment (BarAgam) [reply app/src/test/java/com/teamagam/gimelgimel/app/integration/interactors/DisplayVectorLayerInteractorTest.java]: Yeap

File: data/src/main/java/com/teamagam/gimelgimel/data/layers/LayersLocalCacheData.java (hunk 2 above)
Comment (BarAgam) [reply data/src/main/java/com/teamagam/gimelgimel/data/layers/LayersLocalCacheData.java]: Solid

File: domain/src/main/java/com/teamagam/gimelgimel/domain/messages/SendMessageInteractor.java (hunk 6 above)
Comment (BarAgam) [reply domain/src/main/java/com/teamagam/gimelgimel/domain/messages/SendMessageInteractor.java:42]: Good question....

File: domain/src/main/java/com/teamagam/gimelgimel/domain/layers/LoadAllCachedLayersInteractor.java (hunk 5 above)
Comment (BarAgam) [reply domain/src/main/java/com/teamagam/gimelgimel/domain/layers/LoadAllCachedLayersInteractor.java:43]: EHEUMHMUMH okie.

Comment (teamagam5) [reply domain/src/main/java/com/teamagam/gimelgimel/domain/layers/LoadAllCachedLayersInteractor.java:43]: I knew you're gonna like the idea of a new static method <3

**Pull request merged**